source/iface/integer-functions.rst

```diff
@@ -7,3 +7,854 @@
+``GenInt2`` must be the same as ``GenInt1``; and
+
+If ``GenInt1`` is ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` must also be ``vec`` or the ``__swizzled_vec__`` type,
+and both must have the same element type and the same number
+of elements.
+
+Effects: Computes ``x * y`` and returns the high half of the
+product of ``x`` and ``y``.
+
+Returns: When the inputs are scalars, returns the high half of
+the product of ``x * y``. Otherwise, returns the high half of
+the product of ``x[i] * y[i]`` for each element of
+``x`` and ``y``.
+
+The return type is ``GenInt1`` unless ``GenInt1`` is the
+``__swizzled_vec__`` type, in which case the return type
+is the corresponding ``vec``.
+
+``rotate``
+==========
+
+::
+
+  template<typename GenInt1, typename GenInt2>
+  /*return-type*/ rotate(GenInt1 v, GenInt2 count);
+
+Constraints: Available only if all of the following conditions are met:
+
+``GenInt1`` is a generic integer type as defined above;
+
+If ``GenInt1`` is not ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` must be the same as ``GenInt1``; and
+
+If ``GenInt1`` is ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` must also be ``vec`` or the ``__swizzled_vec__`` type,
+and both must have the same element type and the same number
+of elements.
+
+Effects: For each element in ``v``, the bits are shifted left by the number
+of bits given by the corresponding element in ``count`` (subject to usual
+shift modulo rules described in the OpenCL 1.2 specification
+|SYCL_SPEC_OPENCL|). Bits shifted off the left side of the element are
+shifted back in from the right.
+
+Returns: When the inputs are scalars, the result of rotating
+``v`` by ``count``  as described above. Otherwise, the result of
+rotating ``v[i]`` by ``count[i]`` for each element of ``v``
+and ``count``.
+
+The return type is ``GenInt1`` unless ``GenInt1`` is the
+``__swizzled_vec__`` type, in which case the return type
+is the corresponding ``vec``.
+
+``sub_sat``
+===========
+
+::
+
+  template<typename GenInt1, typename GenInt2>
+  /*return-type*/ sub_sat(GenInt1 x, GenInt2 y);
+
+Constraints: Available only if all of the following conditions are met:
+
+``GenInt1`` is a generic integer type as defined above;
+
+If ``GenInt1`` is not ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` must be the same as ``GenInt1``; and
+
+If ``GenInt1`` is ``vec`` or the ``__swizzled_vec__`` type, then
+``GenInt2`` must also be ``vec`` or the ``__swizzled_vec__`` type,
+and both must have the same element type and the same number
+of elements.
+
+Returns: When the inputs are scalars, returns ``x - y``.
+Otherwise, returns ``x[i] - y[i]`` for each element of ``x``
+and ``y``. The subtraction operation saturates the result.
+
+The return type is ``GenInt1`` unless ``GenInt1`` is the
+``__swizzled_vec__`` type, in which case the return type
+is the corresponding ``vec``.
+
+``upsample``
```

Comment:
I think this is missing the "overload" subheadings to be consistent with other sections

Reply:
do you mean like this : https://github.com/KhronosGroup/SYCL_Reference/pull/163/commits/b0d149f62272efba93cc0dc76e5c1869d8af3823

Reply:
Yes, I think that solves it